prow-configurations/create-gcp-secrets.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+usage () {
+    echo "Provide correct GCP bucket name, keyring and encryption key!"
+    exit 1
+}
+
+CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+SECRET_NAME="gcp-prow"
+FILES=("sa-gke-kyma-integration")
+EXTENSTION="encrypted"
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+    key="$1"
+
+    case ${key} in
+        --bucket)
+            BUCKET=$2
+            shift # past argument
+            shift # past value
+        ;;
+        --keyring)
+            KEYRING=$2
+            shift # past argument
+            shift # past value
+        ;;
+        --key)
+            KEY=$2
+            shift # past argument
+            shift # past value
+        ;;
+        --location)
+            LOCATION=$2
+            shift # past argument
+            shift # past value
+        ;;
+        *)    # unknown option
+            POSITIONAL+=("$1") # save it in an array for later
+            shift # past argument
+        ;;
+    esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+if [[ -z "${BUCKET}" ]] || [[ -z "${KEYRING}" ]] || [[ -z "${KEY}" ]] || [[ -z "${LOCATION}" ]]; then
+    usage
+fi
+
+##########
+
+gsutil acl get gs://${BUCKET} > /dev/null
+
+##########
+
+TMP_DIR=`mktemp -d "${CURRENT_DIR}/temp-XXXXXXXXXX"`
+trap "rm -rf ${TMP_DIR}" EXIT
+
+for FILE in "${FILES[@]}"
+do
+    ENCRYPTED_FILE="${FILE}.${EXTENSTION}"
+    gsutil cp gs://${BUCKET}/${ENCRYPTED_FILE} ${TMP_DIR}/${ENCRYPTED_FILE}
+    gcloud kms decrypt --location "${LOCATION}" --keyring "${KEYRING}" --key "${KEY}" --ciphertext-file "${TMP_DIR}/${ENCRYPTED_FILE}" --plaintext-file "${TMP_DIR}/${FILE}"
+    rm ${TMP_DIR}/${ENCRYPTED_FILE}
+done
+
+##########
+
+if [ "$(ls -A ${TMP_DIR})" ]; then
+    kubectl create secret generic ${SECRET_NAME} --from-file=${TMP_DIR}
```

Comment:
Will we have one secret for all files? I guess I would prefer to have one secret per one file stored in GCS. I can imagine having a file for slack token and for victorops token and I don't want to have them both in one k8s secret cause I might have 2 prow jobs: one for integration with slack and another for victorops.

Reply:
good point, will be changed.